File: hardware/src/main/java/com/satochip/CardListener.java
package com.satochip;

/**
 * Listener for card connection events.
 */
public interface CardListener {
  /**
   * Executes when the card channel is connected.
   *
   * @param channel the connected card channel
   */
  void onConnected(CardChannel channel);

  /**
   * Executes when a previously connected card is disconnected.
   */
  void onDisconnected();
}

File: common/src/androidMain/kotlin/com/blockstream/common/devices/CardListener.kt
package com.blockstream.common.devices

/**
 * Listener for card connection events.
 */
interface CardListener {
    /**
     * Executes when the card channel is connected.
     *
     * @param channel the connected card channel
     */
    fun onConnected(channel: CardChannel)

    /**
     * Executes when a previously connected card is disconnected.
     */
    fun onDisconnected()
}